python/test/gluon/test_lowerings.py

```diff
@@ -255,16 +251,13 @@ def kernel(x_ptr, z_ptr, M: ttgl.constexpr, B: ttgl.constexpr, src_layout: ttgl.
         ttgl.store(z_ptr + z_offs, h)
 
     if src_layout == "linear_layout":
-        if M == 32:
-            src_layout = ttgl.DistributedLinearLayout(
-                reg_bases=[],
-                lane_bases=[[0], [16], [4], [2], [1]] + [[0]] * (THREADS_PER_WARP >> 6),
-                warp_bases=[[0], [8]],
-                block_bases=[],
-                shape=(M, ),
-            )
-        else:
-            pytest.skip("Linear layout is specialized for 32 elements")
+        src_layout = ttgl.DistributedLinearLayout(
```

Comment:
This is wrong. An LL must be created with a given shape

Reply:
See

```
LinearEncodingAttr toLinearEncoding(DistributedEncodingTrait layout,
                                    ArrayRef<int64_t> shape);
```

Reply:
I'm not sure what you mean? This is a 32- or 64-element linear layout depending on how many threads there are in a warp

Reply:
This is `src_layout` so the shape must be (M,) here

Reply:
This layout is wrong as it didn't define reg_bases properly

Reply:
For this you can add the relevant bases to the register bases as needed.